lib/cursor.js

```diff
@@ -788,7 +730,48 @@ function initializeCursor(cursor, callback) {
 
       // Return callback
       done(result);
-    });
+    };
+
+    if (cursor.cmd.find != null) {
```

Comment:
Beware reader, you may find this distasteful (I sure do!), but do not despair. This check was originally done at the `WireProtocol` level, so this merely pushes this anti-pattern closer to the surface and makes it very clear that redesign of the command/operation layer is required.

Much like disambiguating `query` from `command` is the right thing to do, cursor should not act upon all commands equally. A command cursor should use `command`, and a find cursor should use `query`, but none of this is possible for some time so this will have to do for now.

Reply:
Is it worth doing:

```
cursor.query = cursor.server.wireProtocolHandler[cursor.cmd.find != null ? 'query' : 'command'](
         cursor.bson,
         cursor.ns,
         cursor.cmd,
         cursor.cursorState,
         cursor.topology,
         cursor.options
);
```

?

Reply:
I kind of like that it's big and ugly and repetitive, it makes you _want_ to fix it. Rather than fix it here though, I propose we keep pushing it up the stack so that eventually the correct cursor is using the correct methods (`CommandCursor`, `AggregateCursor` both use `command`, regular `Cursor` defaults to `query`)

Reply:
It could even be that `Cursor` has a `method` property that defaults to `query`, and the `CommandCursor`, and `AggregateCursor` both pass in `{ method: command }`